x-pack/platform/plugins/shared/lens/public/react_embeddable/initializers/initialize_internal_api.ts

```diff
@@ -73,21 +70,13 @@ export function initializeInternalApi(
     apiPublishesESQLVariables(parentApi) ? parentApi.esqlVariables$ : []
   );
 
-  const query = initialState.attributes.state.query;
-
-  const panelEsqlVariables$ = new BehaviorSubject<ESQLControlVariable[]>([]);
-  esqlVariables$.subscribe((newVariables) => {
```

Comment:
This led to a memory leak as it wasn't removed at `unmount`.
The code has moved into the `data_loader` where the subscription is correctly removed at `umount`.